open-ce/utils.py

```diff
@@ -103,3 +103,33 @@ def parse_arg_list(arg_list):
 def remove_version(package):
     '''Remove conda version from dependency.'''
     return package.split()[0].split("=")[0]
+
+def make_schema_type(data_type,required=False):
+    '''Make a schema type tuple.'''
+    return (data_type, required)
+
+def validate_type(value, schema_type):
+    '''Validate a single type instance against a schema type.'''
+    if isinstance(schema_type, dict):
+        validate_dict_schema(value, schema_type)
+    else:
+        if not isinstance(value, schema_type):
+            raise OpenCEError("{} is not of expected type {}".format(value, schema_type))
+
+def validate_dict_schema(dictionary, schema):
+    '''Recursively validate a dictionary's schema.'''
+    for k, schema_item in schema.items():
```

Comment:
Is it possible to do `for k, (required, schema_type)`?

Reply:
Ooo maybe. I'll try that.

Reply:
IT worked, so I added this.